packages/data-service/src/graphqlObservable.js

```diff
@@ -0,0 +1,92 @@
+import { Observable } from "rxjs/Observable";
+
+// TODO: refactor this once it has a more compreehensive implementation
+// of the graphql api
+
+// WARNING: This is NOT a spec complete graphql implementation
+// https://facebook.github.io/graphql/October2016/
+
+// eslint-disable-next-line import/prefer-default-export
+export const graphqlObservable = (doc, schema, context) => {
+  const translateOperation = {
+    query: "Query"
+  };
+
+  const resolveStep = (typeMap, definition, context, parent) => {
+    if (definition.kind === "OperationDefinition") {
```

Comment:
As discussed we could do a `pipe` here and test all the steps independently

Reply:
I love the idea, but can we postpone that one when we do this?
https://jira.mesosphere.com/browse/DCOS-22007